### PR TITLE
Jbeorse/refactor ui

### PR DESCRIPTION
**Pull request by jbeorse**
Clean up and refactor front end code. Add templating.

### CODE REVIEW

File: lemongrab/static/js/todo.js [hunk 1]
```diff
@@ -1,132 +1,118 @@
...
+    _bind_events();
+
+  });
 
-  // Create the task html and add it to the To Do list
+  // Create the task and add it to the To Do list
   var _createTask = function(id, description) {
-    var parent = $(constants.incompleteList),
-        task, control;
+    var parent  =   $("#task-list"),
```
Comment (mtkp) [inline lemongrab/static/js/todo.js]: any particular reason for the spacing here?

File: lemongrab/static/js/todo.js [hunk 2]
```diff
@@ -1,132 +1,118 @@
...
-      "type" : "button",
-      "id" : constants.removeAction,
-      "value": constants.removeText,
-      "click": function() {
-        TODO.deleteTask(id);
-      }
-    }).appendTo(control);
+  // Mark task as completed
+  var _completeTask = function(id) {
+    var task  = $("#task-" + id);
```
Comment (mtkp) [inline lemongrab/static/js/todo.js]: same here re spacing

File: lemongrab/static/js/todo.js [hunk 3]
```diff
@@ -1,132 +1,118 @@
...
+  }
 
-  // Retrieve a task from the To Do list and move it to
-  // the complete list.
-  var _moveToComplete = function(id) {
-    var newParent = $(constants.completeList),
-        task = $("#" + constants.taskIdPrefix + id);
+  // Redo a task
+  var _redoTask = function(id) {
+    var task  = $("#task-" + id);
```
Comment (mtkp) [inline lemongrab/static/js/todo.js]: same here re spacing

File: lemongrab/templates/index.html [hunk 4]
```diff
@@ -22,20 +25,28 @@
...
 
-      <div class="task-area" id="complete">
-        <h3>Completed</h3>
+    <!-- Template for tasks to display in the list above -->
+    <script id="task-template" type="text/x-handlebars-template">
+      <div class="task" id="task-{{id}}">
+        <div class="task-body">{{description}}</div>
+        <div class="task-incomplete">Not completed</div>
+        <div class="task-control">
+          <input type="button" id="toggle" value="Complete"
```
Comment (mtkp) [inline lemongrab/templates/index.html:43]: Not sure how much it matters but you could end these inputs with a slash like on line 29,  `<input />`

File: lemongrab/static/js/todo.js (hunk 1 above)
Comment (jbeorse) [inline lemongrab/static/js/todo.js]: The style guide I mentioned had a tab before and after the "=" for all variables. We're going to be declaring every variable with a single var statement, so we're trying to mak it visuall easy to scan the full list of them. The tabs are to align the variable names and instantiations vertically. Of course, I don't have any methods that have multiple variables instantiated at the top. 

Style of course is a preference and we don't need to use that. Or we could only apply it when there are multiple variable instantiations. Would you prefer that?

File: lemongrab/templates/index.html (hunk 4 above)
Comment (jbeorse) [inline lemongrab/templates/index.html:43]: Good catch.

In HTML5 the closing '/' is depracated and shouldn't be used for void elemets. It's supported in all browsers and won't cause a problem, but its really an ignored syntax error. So this instance is correct and the others are wrong. Here's google's style guide's statement on the matter: http://google-styleguide.googlecode.com/svn/trunk/htmlcssguide.xml?showone=Document_Type#Document_Type

I'll fix them.

**Comment by jbeorse** [comment]
I updated the var declaration syntax as mentioned in your comments. I guess that made github hide them all. Anyway, I changed it to be the case that "=" are aligned past the end of the longest variable name. Let me know if you don't like this style.

File: lemongrab/templates/index.html (hunk 4 above)
Comment (mtkp) [inline lemongrab/templates/index.html:43]: Wow... and here I thought we were supposed to be closing them. Good to know!

**Comment by mtkp** [comment]
Looks good Jeff. I was just curious as to why it was formatted as such. I couldn't see the pattern. But I don't have a problem either way.

**Comment by jbeorse** [comment]
No its good, I definitely want to hear it if something is unclear. I'm hoping the code is more readable even to someone without a deeper understanding of what's going on.

**Pull request merged**